modules/programs/piston-cli.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.piston-cli;
+  yamlFormat = pkgs.formats.yaml { };
+in {
+  meta.maintainers = with maintainers; [ ethancedwards8 ];
+
+  options.programs.piston-cli = {
+    enable = mkEnableOption "piston-cli, code runner";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.piston-cli;
+      defaultText = literalExample "pkgs.piston-cli";
+      description = "The piston-cli package to use.";
+    };
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      example = literalExample ''
+        {
+          theme = "emacs";
+          box_style = "MINIMAL_DOUBLE_HEAD";
+          prompt_continuation = "...";
+          prompt_start = ">>>";
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/piston-cli/config.yml</filename>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    xdg.configFile."piston-cli/config.yml" = mkIf (cfg.settings != { }) {
+      source = yamlFormat.generate "config.yml" cfg.settings;
+    };
```

Comment:
Does Piston support Darwin? If so, where does the config go?

Reply:
Yes, and it goes in the same place as linux.